worker/modelworkermanager/modelworkermanager.go

```diff
@@ -257,6 +261,16 @@ func (m *modelWorkerManager) starter(cfg NewModelConfig) func() (worker.Worker,
 		modelName := fmt.Sprintf("%q (%s)", cfg.ModelName, cfg.ModelUUID)
 		m.config.Logger.Debugf("starting workers for model %s", modelName)
 
+		// Get the controller config for the model worker so that we correctly
+		// handle the case where the controller config changes between model
+		// worker restarts.
+		ctx := m.catacomb.Context(context.Background())
+		controllerConfig, err := m.config.ControllerConfigGetter.ControllerConfig(ctx)
+		if err != nil {
+			return nil, errors.Annotate(err, "unable to get controller config")
+		}
+		cfg.ControllerConfig = controllerConfig
+
```

Comment:
Notice how this was only retrieved at the beginning of the loop and not every time the model starts. This controller config was used for log rotation https://github.com/juju/juju/blob/9589eb7adc26ea8010dd968284476ca72ff4a009/cmd/jujud/agent/machine.go#L1039-L1040

Reply:
Additional work will be required to watch the controller config to actually update these values. This PR won't resolve that issue.

Reply:
I know! This is why I was a bit concerned that there is a latent bug here. We should come back around to fix it at some point.